src/core/Akka/Actor/ActorCell.Children.cs

```diff
@@ -374,6 +386,29 @@ private IInternalActorRef MakeChild(Props props, string name, bool async, bool s
             actor.Start();
             return actor;
         }
+
+        private void CheckCreatorsSerialization(Serialization.Serialization ser, object argument)
+        {
+            var objectType = argument.GetType();
+            var serializer = ser.FindSerializerFor(objectType);
+            var bytes = serializer.ToBinary(objectType);
+            var manifestSerializer = serializer as SerializerWithStringManifest;
+            if (manifestSerializer != null)
+            {
+                var manifest = manifestSerializer.Manifest(objectType);
+                if (ser.Deserialize(bytes, manifestSerializer.Identifier, manifest) == null)
+                {
+                    throw new Exception();
```

Comment:
Make both of these `Exception`s being thrown `ArgumentException`s and provide a descriptive error message, similar to what the JVM does.

```scala
case NonFatal(e) ⇒ throw new IllegalArgumentException(s"pre-creation serialization check failed at [${cell.self.path}/$name]", e)
```

Just to make it easier to help debug later.